backend/src/main/java/com/redhat/cloud/notifications/routers/EndpointResource.java

```diff
@@ -211,7 +217,11 @@ public Endpoint createEndpoint(@Context SecurityContext sec,
          * for more details.
          */
         if (this.featureFlipper.isSourcesUsedAsSecretsBackend()) {
-            this.secretUtils.createSecretsForEndpoint(endpoint);
+            try {
+                this.secretUtils.createSecretsForEndpoint(endpoint);
+            } catch (final SourcesException e) {
+                Log.error(e.getMessage(), e);
+            }
```

Comment:
Here, we are unable to store a secret in Sources during an endpoint creation. I think the entire endpoint creation should fail in that case, resulting in a 500 status for the caller. We would still need the error log entry.

Reply:
Good call, I agree, yes. I was thinking on returning a generic error message like "Unable to create the integration due to an internal error" and that's it, since the user might not be interested in knowing anything about the secrets back end. Same for the apps.

What do you think?

Reply:
Sounds good to me

Reply:
Done!